tests/Ontology/TWAnySignerTests.cpp

```diff
@@ -190,4 +191,45 @@ TEST(TWAnySingerOntology, OngWithdraw) {
         rawTx);
 }
 
+TEST(TWAnySingerOntology, Oep4Decimal) {
+    auto input = Proto::SigningInput();
+    input.set_contract("ff31ec74d01f7b7d45ed2add930f5d2239f7de33");
+    input.set_method("decimals");
+    input.set_nonce(0x1234);
+    auto data = Oep4TxBuilder::build(input);
+    auto rawTx = hex(data);
+    EXPECT_EQ("00d1341200000000000000000000000000000000000000000000000000000000000000000000000000002000c108646563696d616c736733def739225d0f93dd2aed457d7b1fd074ec31ff0000", rawTx);
+}
+
+TEST(TWAnySingerOntology, Oep4BalanceOf) {
+    auto input = Proto::SigningInput();
+    input.set_contract("ff31ec74d01f7b7d45ed2add930f5d2239f7de33");
+    input.set_method("balanceOf");
+    input.set_query_address("AeaThtPwh5kAYnjHavzwmvxPd725nVTvbM");
+    input.set_nonce(0x1234);
+    auto data = Oep4TxBuilder::build(input);
+    auto rawTx = hex(data);
+    EXPECT_EQ("00d1341200000000000000000000000000000000000000000000000000000000000000000000000000003614fa2254ffaee3c3e1172e8e98f800e4105c74988e51c10962616c616e63654f666733def739225d0f93dd2aed457d7b1fd074ec31ff0000", rawTx);
+}
+
+TEST(TWAnySingerOntology, Oep4Transfer) {
+    auto ownerPrivateKey =
+        parse_hex("4646464646464646464646464646464646464646464646464646464646464652");
+    auto payerPrivateKey =
+        parse_hex("4646464646464646464646464646464646464646464646464646464646464646");
+    auto input = Proto::SigningInput();
+    input.set_contract("ff31ec74d01f7b7d45ed2add930f5d2239f7de33");
+    input.set_method("transfer");
+    input.set_owner_private_key(ownerPrivateKey.data(), ownerPrivateKey.size());
+    input.set_payer_private_key(payerPrivateKey.data(), payerPrivateKey.size());
+    input.set_to_address("AVY6LfvxauVQAVHDV9hC3ZCv7cQqzfDotH");
+    input.set_amount(233);
+    input.set_gas_price(2500);
+    input.set_gas_limit(50000);
+    input.set_nonce(0x1234);
+    auto data = Oep4TxBuilder::build(input);
```

Comment:
You need to use `TWAnySignerSign` method here

Reply:
thank you for pointing out, updated!